lyndon/lyndon_factorization.py

```diff
@@ -0,0 +1,16 @@
+import operator
+
+def duval(text, n, less = operator.__lt__):
+  i = 1
+  while i <= n:
+    j, k = i + 1, i
+    while j <= n and not less(text[j], text[k]):
+      if less(text[k], text[j]):
+        k = i
+      else:
+        k += 1
```

Comment:
In this case maybe "k = i if less(text[k], text[j]) else k + 1" could be even better